spec/system/managing_topics_spec.rb

```diff
@@ -21,4 +21,18 @@
     expect(current_path).to eq topics_path
     expect(page).to have_content 'tell us a story'
   end
+
+  scenario 'viewing a topic' do
+    visit topics_path
+    click_on('What is your favourite colour')
+    expect(current_path).to eq topic_path(@topic.id)
+    expect(page).to have_content('Topic: 1')
+  end
+
+  scenario 'when deleting a topic' do
+    visit topic_path(@topic.id)
+    click_on 'Delete Topic'
```

Comment:
This is a great feature spec: it focuses on what the user can see. I'd say it's fine as it is, but as it presents an opportunity for teaching something new...

A possible enhancement, if you wanted to check that the action definitely deleted the topic, would be:
```
expect {
  click_on 'Delete Topic'
}.to change(Toppic, :count).by(-1)
```

Notice that the block being passed to expect here: They're curly brackets instead of the usual brackets. You might want to consider what's going on from a Ruby perspective 🤔 .

Reply:
```
 Failure/Error: expect {accept_confirm { click_on 'Delete Topic' }}.to change(Topic, :count).by(-1)
       expected #count to have changed by -1, but was changed by 0
```

Reply:
think I might be nesting too much there.... it works if I turn the UJS confirm message off and remove the accept_confirm bit